proto/redpanda/api/console/v1alpha1/security.proto

```diff
@@ -0,0 +1,236 @@
+message ListRolesResponse {
+  // The roles in the system.
+  repeated Role roles = 1;
+
+  // Token to retrieve the next page.
+  string next_page_token = 2;
+}
+
+// CreateRoleRequest is the request for CreateRole.
+message CreateRoleRequest {
+  // The role to create.
+  Role role = 1;
+}
+
+// CreateRoleResponse is the response for CreateRole.
+message CreateRoleResponse {
+  // The role.
+  Role role = 1;
+}
+
+// CreateRoleRequest is the request for CreateRole.
+message GetRoleRequest {
+  // The role name.
+  string role_name = 1 [
+    (buf.validate.field).required = true,
+    (buf.validate.field).string.min_len = 1,
+    (buf.validate.field).string.max_len = 128,
+    (buf.validate.field).string.pattern = "^[a-zA-Z0-9-_]+$"
+  ];
+}
+
+// GetRoleResponse is the response to GetRole.
+message GetRoleResponse {
+  // The Role.
+  Role role = 1;
+
+  // Members assigned to the role.
+  repeated RoleMembership members = 2;
+}
+
+// UpdateRoleRequest is the request for UpdateRole.
+message UpdateRoleRequest {
+  // The role name.
+  string role_name = 1 [
+    (buf.validate.field).required = true,
+    (buf.validate.field).string.min_len = 1,
+    (buf.validate.field).string.max_len = 128,
+    (buf.validate.field).string.pattern = "^[a-zA-Z0-9-_]+$"
+  ];
+
+  // The updated role.
+  Role role = 2;
+}
+
+// UpdateRoleResponse is the response for UpdateRole.
+message UpdateRoleResponse {
+  // The role.
+  Role role = 1;
+}
+
+// DeleteRoleRequest is the request for DeleteRole.
+message DeleteRoleRequest {
+  // The role name.
+  string role_name = 1 [
+    (buf.validate.field).required = true,
+    (buf.validate.field).string.min_len = 1,
+    (buf.validate.field).string.max_len = 128,
+    (buf.validate.field).string.pattern = "^[a-zA-Z0-9-_]+$"
+  ];
+
+  // Whether to delete the ACLs bound to the role.
+  optional bool delete_acls = 2;
+}
+
+// DeleteRoleResponse is the response for DeleteRole.
+message DeleteRoleResponse {}
+
+// List role members for a role. That is user principals assigned to that role.
+message ListRoleMembersRequest {
+  // Filter options.
+  message Filter {
+    // Filter results to only members with names which contain the string.
+    string name_contains = 1 [(buf.validate.field).string.max_len = 128];
+  }
+
+  // The role name.
+  string role_name = 1 [
+    (buf.validate.field).required = true,
+    (buf.validate.field).string.min_len = 1,
+    (buf.validate.field).string.max_len = 128,
+    (buf.validate.field).string.pattern = "^[a-zA-Z0-9-_]+$"
+  ];
+
+  // Optional filter.
+  optional Filter filter = 2;
+
+  // Page size.
+  int32 page_size = 3 [(buf.validate.field).int32 = {
+    gte: -1,
+    lte: 1000
+  }];
+
+  // Value of the next_page_token field returned by the previous response.
+  // If not provided, the system assumes the first page is requested.
+  string page_token = 4;
+}
+
+// ListRoleMembersResponse is the response for ListRoleMembers.
+message ListRoleMembersResponse {
```

Comment:
Wondering if it's worth to include the role name here again?

Reply:
Added `string role_name = 1;` to the response message.